packages/malloy-render/src/component/render-result-metadata.ts

```diff
@@ -21,283 +22,1562 @@
 import {mergeVegaConfigs} from './vega/merge-vega-configs';
 import {baseVegaConfig} from './vega/base-vega-config';
 import {renderTimeString} from './render-time';
 import {generateBarChartVegaSpec} from './bar-chart/generate-bar_chart-vega-spec';
-import {createResultStore} from './result-store/result-store';
+import {createResultStore, ResultStore} from './result-store/result-store';
 import {generateLineChartVegaSpec} from './line-chart/generate-line_chart-vega-spec';
-import {parse, Config} from 'vega';
-
-function createDataCache() {
-  const dataCache = new WeakMap<DataColumn, QueryData>();
-  return {
-    get: (cell: DataColumn) => {
-      if (!dataCache.has(cell) && cell.isArray()) {
-        const data: DataRowWithRecord[] = [];
-        const fields = cell.field.allFields;
-        for (const row of cell) {
-          const record = {__malloyDataRecord: row};
-          for (const field of fields) {
-            const value = row.cell(field).value;
-            // TODO: can we store Date objects as is? Downstream chart code would need to be updated
-            record[field.name] =
-              value instanceof Date ? value.valueOf() : row.cell(field).value;
-          }
-          data.push(record as DataRowWithRecord);
-        }
-        dataCache.set(cell, data);
-      }
-      return dataCache.get(cell)!;
-    },
-  };
-}
+import {parse, Config, Runtime} from 'vega';
+import * as Malloy from '@malloydata/malloy-interfaces';
 
 export type GetResultMetadataOptions = {
   getVegaConfigOverride?: VegaConfigHandler;
 };
 
 export function getResultMetadata(
-  result: Result,
+  result: Malloy.Result,
   options: GetResultMetadataOptions = {}
 ) {
-  const fieldKeyMap: WeakMap<Field | Explore, string> = new WeakMap();
-  const getCachedFieldKey = (f: Field | Explore) => {
-    if (fieldKeyMap.has(f)) return fieldKeyMap.get(f)!;
-    const fieldKey = getFieldKey(f);
-    fieldKeyMap.set(f, fieldKey);
-    return fieldKey;
-  };
+  const fields: Malloy.DimensionInfo[] = [];
+  for (const field of result.schema.fields) {
+    if (field.kind === 'dimension') {
+      fields.push(field);
+    }
+  }
 
-  const dataCache = createDataCache();
-  const rootField = result.data.field;
-  const metadata: RenderResultMetadata = {
-    fields: {},
-    fieldKeyMap,
-    getFieldKey: getCachedFieldKey,
-    field: (f: Field | Explore) => metadata.fields[getCachedFieldKey(f)],
-    getData: dataCache.get,
-    modelTag: result.modelTag,
-    resultTag: result.tagParse().tag,
-    rootField,
-    store: createResultStore(),
-  };
+  const rootFieldMeta = new RootField(
+    {
+      name: 'root',
+      type: {
+        kind: 'array_type',
+        element_type: {
+          kind: 'record_type',
+          fields,
+        },
+      },
+      annotations: result.annotations,
+    },
+    {
+      modelTag: tagFromAnnotations(result.model_annotations, '## '),
+      store: createResultStore(),
+      queryTimezone: result.query_timezone,
```

Comment:
Should this just be in a `#(malloy) ` tag?

Reply:
Most of the other miscellaneous metadata from running a query is dumped into `#(malloy) ` annotations. Should timezone be a part of the query response, or a tag on the query?